src/artifact_providers/base.ts

```diff
@@ -118,15 +121,31 @@ export abstract class BaseArtifactProvider {
   }
 
   /**
-   * Downloads the given artifact file (without caching)
+   * Downloads the given file from the artifact store.
+   *
+   * This method's caller caches downloaded files during the instance's
+   * lifetime, so this method should only be called once per artifact
+   *
+   * @param artifact An artifact object to download
+   * @param downloadDirectory Directory where downloaded artifact is stored
+   * @returns Absolute path to the saved file
    */
   protected abstract async doDownloadArtifact(
     artifact: CraftArtifact,
     downloadDirectory: string
   ): Promise<string>;
 
   /**
-   * Downloads multiple artifacts to the given directory
+   * Given an arry of artifacts, returns an arry of local paths to the those
```

Comment:
```suggestion
   * Given an array of artifacts, returns an array of local paths to those
```

Reply:
Damn, and I proofread and everything. Thanks for catching!